### DISCUSSION TITLE
Is it possible to access an SQL database?

### DISCUSSION THREAD

**Discussion by lstillwe** [TerriaJS/terriajs#5077] [Q&A]
Is it possible to access an SQL database - i.e. to get info to make menu items, etc using the terriajs server? Any example of someone who has done this? Is it better to customize the terriajs server, or use a different server? Thanks!

**Comment by geospatialmapmashup** [comment]
We've used MS SQL Server Geometry Blob and used OGR2OGR or other tools to request GeoJSON and show that on map.
I don't believe MS SQL Server has like what postgis has ST_asGEOJSON 

This articile shows how to get GeoJSON output
https://techcommunity.microsoft.com/t5/sql-server/returning-spatial-data-in-geojson-format-part-2/ba-p/384610
https://docs.microsoft.com/en-us/sql/t-sql/spatial-geometry/stastext-geometry-data-type?view=sql-server-ver15

Keep your data small to a bbox
and limit the number of attributes columns/fields so part of your select query select important columns
keep geojson under 1.5mb so you don't crash the map/browser

**Reply by lstillwe** [reply]
I wasn't specifically talking about spatial data in the database, but instead metadata to display in popups or other parts of the dashboard. Would that be done through React?

**Comment by tephenavies** [comment]
Hi @lstillwe. I think forking, adding a route to terriajs-server and putting the code in `lib/controllers` would be a fairly good way of accomplishing this, but if you want to avoid modifying terriajs-server you could make a separate server.
